### ISSUE TITLE
Scroll does not work properly when rows are EXPANDED

### ISSUE THREAD

**Issue by unkarjedy** [angular-ui/ui-grid#5211]
The bug is quite simple to reproduce.

Check first example:
[]( http://ui-grid.info/docs/#/tutorial/216_expandable_grid)

1) Expand first 3 rows
2) Scroll down to 200 row

There is a gap in the bottom... It seams that table data height calculation ignores expanded rows height when scrolling.

**Comment by scottwo** [comment]
This is being worked on in #4508. Hopefully, that linked correctly.

**Comment by vance** [comment]
Indeed, I can reproduce this with only one row, when the row height is larger than the screen, it JUMPS ahead instead of showing the bottom of the row template (in this case, there's nothing there, but it skips showing the end). 

**Heres a plunkr demonstrating it. **
[http://plnkr.co/edit/kLMMl74UjRXiuS3h0s8x?p=preview](http://plnkr.co/edit/kLMMl74UjRXiuS3h0s8x?p=preview)

**Comment by vance** [comment]
I have graphed some variables and added the ability to tag events. We can clearly see the the parts where the yellow line (canvasHeight) jumps to a weird value when scrolling past the expanded row. This may be removing and adding it to the rowCache... but as you can see, there's two values it jumps to, so i'm thinking maybe it's both hitting and not hitting the `self.canvasHeightShouldUpdate`

<img width="1381" alt="screen shot 2016-05-04 at 11 08 57 am" src="https://cloud.githubusercontent.com/assets/145471/15024175/b11b2d24-11e8-11e6-8b64-4810a9bf6cea.png">

more work on this today.

**Comment by vance** [comment]
I'm going nuts here... anyone know where the visibleRowCache is actually set? All I can find are references getting the value, but not setting it. Maybe i'm blind.

**Comment by vance** [comment]
Good news and bad news. Good news is, for basic cases, I have a solution for the expandable row miscalculation. The bad news is, this may be a systemic problem, and does **NOT** fix it if you also use infinite scroll because it has the same flawed assumption that rows are the same height. 


Yay. I fixed half the problem! We need to compute the proper index based on where scrollTop lands compared to the expanded rows' positions AND height.

doing this:
```

var theoreticalRowIndex = 0;
    var currentHeight = 0;
    var i = 0;
    self.visibleRowCache.forEach(function(r){
      if( scrollTop > currentHeight && scrollTop < currentHeight + r.height){
        theoreticalRowIndex = i;
      }
      currentHeight+= r.height;
      i++;
    });
```
instead of

`maxRowIndex * scrollPercentage`

gets rid of the "jumping" at the end of the expandable row.

**Comment by orneryd** [comment]
Thanks for contributing! This repository has been transferred, and we're preparing the new 5.x release. We're closing older issues and pull requests as part of the reset. Please join us on Discord to follow the work or continue the conversation: https://discord.gg/Baz4w8ZWN
